opentracing-brave/src/main/java/io/opentracing/BraveSpanImpl.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.ClientTracer;
+import com.github.kristofa.brave.ServerTracer;
+import com.github.kristofa.brave.SpanId;
+
+
+final class BraveSpanImpl extends AbstractSpan {
+
+    final SpanId spanId;
+    final Optional<Span> parent;
+    final Optional<ServerTracer> serverTracer;
```

Comment:
main thought here (following my comment) is that open tracing span's aren't implicitly propagated, while Brave's are. This is why you need to pass around a ServerTracer or ClientTracer (because the span thread-locals are in there). Knowing that anyone using OT has to do propagation themselves, whatever Brave propagation exists is redundant to that.

Long story short, I think the easiest way is to just yank Brave's Sampler and SpanCollector, manually creating spans, as opposed to passing around the server, client or local tracer.